partner_risk_insurance/tests/test_invoice_credit_policy.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2024 Moduon Team S.L. <info@moduon.team>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import exceptions
+from odoo.tests import Form, tagged
+
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+
+
+@tagged("post_install", "-at_install")
+class TestInvoiceCreditPolicy(AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.policy_state = cls.env["credit.policy.state"].create(
+            {"name": "Insurance Policy", "insure_invoices": True}
+        )
+        cls.policy_company = cls.env["credit.policy.company"].create(
+            {"name": "Insurance Company"}
+        )
+        cls.partner_ins = cls.env["res.partner"].create(
+            {
+                "name": "Test Insurance Partner",
+                "credit_policy_state_id": cls.policy_state.id,
+                "credit_policy_company_id": cls.policy_company.id,
+            }
+        )
+
+    def test_insured_out_invoice(self):
+        # Create an invoice without partner
+        out_inv = self.init_invoice("out_invoice", partner=self.partner_a)
+        out_inv_f = Form(out_inv)
```

Comment:
**suggestion(non-blocking):** When editing an existing record, you can do this:

```suggestion
        with Form(out_inv) as out_inv_f:
```

When you exit the context manager, it is saved automatically.